android/src/main/java/com/media4care/twilio/video/TwilioVideoActivity.java

```diff
@@ -905,6 +927,50 @@ public void onVideoTrackDisabled(
                 RemoteParticipant remoteParticipant,
                 RemoteVideoTrackPublication remoteVideoTrackPublication
             ) {}
+
+            @Override
+            public void onNetworkQualityLevelChanged(@NonNull RemoteParticipant remoteParticipant, @NonNull NetworkQualityLevel networkQualityLevel) {
+                remoteParticipantNQStatus.setText("Remote: " + networkQualityLevel.name());
+            }
+        };
+    }
+
+    private LocalParticipant.Listener localParticipantListener() {
+        return new LocalParticipant.Listener() {
+            @Override
+            public void onNetworkQualityLevelChanged(@NonNull LocalParticipant localParticipant, @NonNull NetworkQualityLevel networkQualityLevel) {
+                localParticipantNQStatus.setText("Local: " + networkQualityLevel.name());
+            }
+
+            @Override
+            public void onAudioTrackPublished(@NonNull LocalParticipant localParticipant, @NonNull LocalAudioTrackPublication localAudioTrackPublication) {
+
```

Comment:
why are these functions needed?

Reply:
Yes, they are part of the interface and required